samples/DS3232RTC_NTP_Setter/app/application.cpp

```diff
@@ -33,34 +32,26 @@ void onNtpReceive(NtpClient& client, time_t timestamp) {
 	Serial.printf("Time synchronized: %s\n", SystemClock.getSystemTimeString().c_str());
 }
 
-void gotIP(IPAddress ip, IPAddress netmask, IPAddress gateway)
+void systemReady()
 {
 	ntpClient.requestTime();
 }
 
-void connectFail(String ssid, uint8_t ssid_len, uint8_t bssid[6], uint8_t reason)
-{
-	debugf("I'm NOT CONNECTED!");
-}
-
 void init()
 {
 	Serial.begin(SERIAL_BAUD_RATE);
-	Serial.systemDebugOutput(true); // Allow debug print to serial
 	Serial.println("Sming DSRTC_NTP_SETTER started!");
-//Uncomment proper line, or set right pins by hand
 	Wire.pins(0, 2); //Change to your SCL - 0,SDA - 2 GPIO pin number
-//	Wire.pins(5, 4); //Change to your SCL - 5,SDA - 4 GPIO pin number
 	Wire.begin();
 
 	// Station - WiFi client
 	WifiStation.config(WIFI_SSID, WIFI_PWD); // Put you SSID and Password here
 	WifiStation.enable(true);
-	WifiEvents.onStationGotIP(gotIP);
-	WifiEvents.onStationDisconnect(connectFail);
 
 	// set timezone hourly difference to UTC
 	SystemClock.setTimeZone(2); // GMT+2
 
+	System.onReady(systemReady);
```

Comment:
> Many examples use onReady to initiate networked operations. 

It depends on your definition of "initiate networked operations".  System.onReady callbacks are  triggered after the TCP stack is ready. So far that should be included in "initiate networked operations". But that is different than connecting to an AP and getting IP address from the DHCP.   

> Are these examples all wrong?

Without knowing which examples you mean I cannot say for sure. Please, post here one or two names of samples that you've been referring to.